WordPress/src/main/java/org/wordpress/android/ui/media/MediaPreviewFragment.java

```diff
@@ -277,43 +278,74 @@ private void loadImage(String mediaUri) {
         }
 
         mImageView.setVisibility(View.VISIBLE);
-        if (mSite == null || SiteUtils.isPhotonCapable(mSite)) {
+        if ((mSite == null || SiteUtils.isPhotonCapable(mSite)) && !UrlUtils.checkIfContentUri(mediaUri)) {
             int maxWidth = Math.max(DisplayUtils.getDisplayPixelWidth(getActivity()),
                     DisplayUtils.getDisplayPixelHeight(getActivity()));
             mediaUri = PhotonUtils.getPhotonImageUrl(mediaUri, maxWidth, 0);
         }
         showProgress(true);
-        mImageManager.loadWithResultListener(mImageView, ImageType.IMAGE, mediaUri, ScaleType.CENTER, null,
-                new RequestListener<Drawable>() {
-                    @Override
-                    public void onResourceReady(@NonNull Drawable resource, @Nullable Object model) {
-                        if (isAdded()) {
-                            // assign the photo attacher to enable pinch/zoom - must come before setImageBitmap
-                            // for it to be correctly resized upon loading
-                            PhotoViewAttacher attacher = new PhotoViewAttacher(mImageView);
-                            attacher.setOnViewTapListener(new PhotoViewAttacher.OnViewTapListener() {
+
+        if (UrlUtils.checkIfContentUri(mediaUri)) {
+            mImageManager
+                    .loadWithResultListener(mImageView, ImageType.IMAGE, Uri.parse(mediaUri), ScaleType.CENTER, null,
```

Comment:
I wonder, can we utilize `Uri.parse(mediaUri)` for non-content URI? I briefly tried it with remote URLs from the free photo library, and it worked ok. If there is any issue with this, do you think we can extract `RequestListener` from here and use it for both `loadWithResultListener` calls?

Reply:
> I wonder, can we utilize Uri.parse(mediaUri) for non-content URI? I briefly tried it with remote URLs from the free photo library, and it worked ok.

Super cool! Thanks for spotting this @khaykov Eliminated all the other concerns I was having and the duplicated code is now gone. I resolved it here 0c024ad I am hiding that it's being converted to a `Uri` in the `ImageManager` because it's still accepting a `String` url. That's fine right? All other methods are accepting `String` values. My only concern with my approach is that I am wondering if someone would use this for a content URI and it works and then attempt to do the same with another method in the `ImageManager` and the behavior is different and the issue isn't easily noticeable. Let me know your thoughts.

Reply:
Yeah, I agree that it might be confusing. We should have the same behavior in other methods too. Changing `imgUrl` from String to Uri without overloading will result in quite a lot of changes, so URI parsing in other methods sounds like the most reasonable approach to me.

Reply:
Hey @khaykov I was going to do this but decided against it since there would be alot of functionality to test/double check to be sure no regression occurs. ( I know it would work but would be worth retesting since this impacts a lot of core functionality) So in the interest of time, I simply created an overload that utilized the `URI` type for the `MediaPreviewFragment` so the rest of the functionality in the `ImageManager` remains untouched. Did that here 1e7df64

Reply:
@jd-alexander Makes sense! Thank you :)